### ISSUE TITLE
Redundant context loads when distinct pshufd's in a block

### ISSUE THREAD

**Issue by alyssarosenzweig** [FEX-Emu/FEX#3657]
bytemark has a sequence like:

```
        "pshufd xmm2,xmm3,0xf5",
        "addpd  xmm1,xmm12",
        "pmuludq xmm3,xmm9",
        "pshufd xmm3,xmm3,0xe8",
```

because the pshufd indices differ, we can't cache the constant. and we end up generating code

```
        "ldr x0, [x28, #1760]",
        "ldr q3, [x0, #3920]",
        "tbl v18.16b, {v19.16b}, v3.16b",
        "fadd v17.2d, v17.2d, v28.2d",
        "uzp1 v4.4s, v19.4s, v19.4s",
        "uzp1 v5.4s, v25.4s, v25.4s",
        "umull v19.2d, v4.2s, v5.2s",
        "ldr x0, [x28, #1760]",
        "ldr q4, [x0, #3712]",
        "tbl v19.16b, {v19.16b}, v4.16b",
```

Notice that we reload the base vector addresses in the same block, instead of caching it. At minimum we should probably fix that. But I'm also questioning if there's maybe a better way to do pshufd? Not sure

**Comment by Sonicadvance1** [comment]
0xf5 (0b11_11_01_01) could be a dup+dup+zip but if we cache the constant I don't think that is any better.
0xe8 (0b11_10_10_00) I don't think there is much that can be done there?

**Comment by Sonicadvance1** [comment]
We also don't optimize the case where all elements are the same aside from one (or two sequential. Which those turn in to a dup+ins. Definitely missing some cases as the comment in the source claims :)
eg:
- 0bAA_AA_AA_BB
- 0bBB_AA_AA_AA
- etc, rotate the next two
- 0bAA_AA_AA_00 - dup + ins
- 0bAA_AA_01_00 - dup + zip/ins
- 0b11_10_AA_AA - dup + zip/ins
- 0bAA_AA_10_01 - dup + ins (zip won't work here)

**Comment by alyssarosenzweig** [comment]
> 0xe8 (0b11_10_10_00) I don't think there is much that can be done there?


The ticket is about the `ldr x0, [x28, #1760]` not getting cached across different pshufds in the same block

**Comment by Sonicadvance1** [comment]
Indeed, I was responding to `But I'm also questioning if there's maybe a better way to do pshufd? `

**Comment by alyssarosenzweig** [comment]
(Alternatively - could/should named vector constants go in the context itself? Or, could we put them at a fixed address? Wondering if we can eliminate the extra indirection)
